smtp.go

```diff
@@ -84,6 +88,19 @@ const (
 	DSNAddressTypeUTF8   DSNAddressType = "UTF-8"
 )
 
+type DeliverByMode string
+
+const (
+	DeliverByNotify DeliverByMode = "N"
+	DeliverByReturn DeliverByMode = "R"
+)
+
+type DeliverByOptions struct {
+	ByTime  int64
+	ByMode  DeliverByMode
+	ByTrace bool
```

Comment:
Nit: I think we can drop the `By` prefix for these fields, since it's already in the struct name?